shopinvader_address/models/res_partner.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, models
+from odoo.exceptions import MissingError, UserError
+
+
+class ResPartner(models.Model):
+
+    _inherit = "res.partner"
+
+    def _shopinvader_billing_addresses_already_used(self):
+        """
+        Check if Address is used on posted account move
+        """
+        self.ensure_one()
+        move_id = (
+            self.env["account.move"]
+            .sudo()
+            .search(
+                [("commercial_partner_id", "=", self.id), ("state", "=", "posted")],
+                limit=1,
+            )
+        )
+        return len(move_id) > 0
+
+    # --- Billing ---
+    # Billing addresses is unique and corresponds to authenticated_partner
+
+    def _get_shopinvader_billing_addresses(self) -> "ResPartner":
+        self.ensure_one()
+        return self
```

Comment:
I'm wondering if we should check that `self is self.commercial_partner_id` in all these methods.

Reply:
Yes or instead of operating on `self`, directly use` self.commercial_partner_id`? 
What should we do if` self != self.commercial_partner_id` ?

Reply:
@sbidoul

Reply:
Maybe add a method `_check_shopinvader_address_partner()` that raises a UserError if self is not self.commercial_partner_id